### PR TITLE
Add NATS Streaming Operator Helm Chart

### PR DESCRIPTION
**Pull request by Starefossen**
This Pull Request adds a Helm chart for the NATS Streaming Operator similar to the NATS Operator (https://github.com/nats-io/nats-operator/tree/master/helm/nats-operator)

All suggestions, comments and improvements are welcome. Also thinking if we should submit this to the official chart repository to make it easy to install without having to git clone the repo.

**TODO:**

* [ ] Document values.yaml in README.md
* [ ] Uodate NOTES.txt

**OTHER:**

* [ ] liveness and readiness for NATS Streaming Operator deployment (blocked on  #31)
* [ ] Cluster Scoped install similar to the NATS Operator?
* [ ] Add nats-operator as a dependency to auto-install NATS Operator?

### CODE REVIEW

**Comment by philicious** [comment]
I like having a Helm Chart. thx @Starefossen !. Will try it out soon as I need to deploy and manage a nats-streaming cluster one way or another

**Comment by tegk** [comment]
Will be this merged?

File: helm/nats-streaming-operator/templates/rbac.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+metadata:
+  name: nats-streaming-operator-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: nats-streaming-operator
+subjects:
+- kind: ServiceAccount
+  name: nats-streaming-operator
+  namespace: default
```
Comment (omenv) [inline helm/nats-streaming-operator/templates/rbac.yaml]: I think the better way to use:
```
namespace: {{.Release.Namespace}}
```

File: helm/nats-streaming-operator/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+              fieldPath: metadata.namespace
+        - name: MY_POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        ports:
+        - name: readyz
+          containerPort: 8080
+        {{- if .Values.livenessProbe.enabled }}
+        livenessProbe:
```
Comment (omenv) [inline helm/nats-streaming-operator/templates/deployment.yaml]: nats-streaming-operator does not have 8080 port into it's pod. So I had a problem with this livenessprobe.

**Comment by omenv** [comment]
Maybe it is a good way to add a new brunch for the helm-chart preparing. 
Also good to see options for persistence/SQL.

**Comment by dellintosh** [comment]
@Starefossen I have an open PR for your fork, if you get a moment to merge that in.  It addresses a few of the concerns brought up, and I'm hoping that we can get this merged in.

https://github.com/Starefossen/nats-streaming-operator/pull/1

**Comment by wallyqs** [comment]
@dellintosh thanks for the contribution, since there are some adopting the helm chart already I think we should merge as is and you can send PR here if you want.

**Comment by wallyqs** [comment]
Thanks a lot @Starefossen for the PR, sorry for the delay in merging... :bow:

**Pull request merged**